common/src/test/java/org/opensearch/ml/common/agent/MLToolSpecTest.java

```diff
@@ -32,11 +32,70 @@ public void writeTo() throws IOException {
         Assert.assertEquals(spec.getParameters(), spec1.getParameters());
         Assert.assertEquals(spec.getDescription(), spec1.getDescription());
         Assert.assertEquals(spec.isIncludeOutputInAgentResponse(), spec1.isIncludeOutputInAgentResponse());
+        Assert.assertEquals(spec.getConfigMap(), spec1.getConfigMap());
+    }
+
+    @Test
+    public void writeToEmptyConfigMap() throws IOException {
+        MLToolSpec spec = new MLToolSpec("test", "test", "test", Map.of("test", "test"), false, Collections.EMPTY_MAP);
+        BytesStreamOutput output = new BytesStreamOutput();
+        spec.writeTo(output);
+        MLToolSpec spec1 = new MLToolSpec(output.bytes().streamInput());
+
+        Assert.assertEquals(spec.getType(), spec1.getType());
+        Assert.assertEquals(spec.getName(), spec1.getName());
+        Assert.assertEquals(spec.getParameters(), spec1.getParameters());
+        Assert.assertEquals(spec.getDescription(), spec1.getDescription());
+        Assert.assertEquals(spec.isIncludeOutputInAgentResponse(), spec1.isIncludeOutputInAgentResponse());
+        Assert.assertEquals(spec.getConfigMap(), spec1.getConfigMap());
+    }
+
+    @Test
+    public void writeToNullConfigMap() throws IOException {
+        MLToolSpec spec = new MLToolSpec("test", "test", "test", Map.of("test", "test"), false, null);
+        BytesStreamOutput output = new BytesStreamOutput();
+        spec.writeTo(output);
+        MLToolSpec spec1 = new MLToolSpec(output.bytes().streamInput());
+
+        Assert.assertEquals(spec.getType(), spec1.getType());
+        Assert.assertEquals(spec.getName(), spec1.getName());
+        Assert.assertEquals(spec.getParameters(), spec1.getParameters());
+        Assert.assertEquals(spec.getDescription(), spec1.getDescription());
+        Assert.assertEquals(spec.isIncludeOutputInAgentResponse(), spec1.isIncludeOutputInAgentResponse());
+        Assert.assertNull(spec1.getConfigMap());
     }
 
     @Test
     public void toXContent() throws IOException {
-        MLToolSpec spec = new MLToolSpec("test", "test", "test", Map.of("test", "test"), false);
+        MLToolSpec spec = new MLToolSpec("test", "test", "test", Map.of("test", "test"), false, Map.of("configKey", "configValue"));
+        XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
+        spec.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        String content = TestHelper.xContentBuilderToString(builder);
+
+        Assert
+            .assertEquals(
+                "{\"type\":\"test\",\"name\":\"test\",\"description\":\"test\",\"parameters\":{\"test\":\"test\"},\"include_output_in_agent_response\":false,\"config\":{\"configKey\":\"configValue\"}}",
```

Comment:
I don't think the order of fields in JSON is guaranteed unless you specify a sorting order.  Same comment in other assertions.

Reply:
Do you have any suggestions?

Reply:
I usually do assertions that the string contains each key value pair.  Or just extract to a map with a util function and check keys.   I think there are some optional arguments with the Xcontent that let you specify an ordering.

Reply:
Looked into this.  Seems internal comments indicate the "XContent" preserves the same ordering, so since you're just parsing the same thing you built, this should be fine in this test case.  The concern would be if you're using any sort of external parsing using a map, where the ordering isn't guaranteed.